main.cpp

```diff
@@ -4999,10 +5000,29 @@ bool encrypt_command::execute(device_map &devices) {
     aes_key_share_t aes_key_share;
     aes_file->seekg(0, std::ios::end);
     if (aes_file->tellg() != 128) {
-        fail(ERROR_INCOMPATIBLE, "The AES key share must be a 128 byte file (the supplied file is %d bytes)", aes_file->tellg());
+        // Generate a random key share from 256-bit key
+        if (aes_file->tellg() != 32) {
+            fail(ERROR_INCOMPATIBLE, "The AES key share must be a 128 byte key share, or a 32 byte key (the supplied file is %d bytes)", aes_file->tellg());
+        }
+        aes_key_t tmp_key;
+        aes_file->seekg(0, std::ios::beg);
+        aes_file->read((char*)tmp_key.bytes, sizeof(tmp_key.bytes));
+
+        std::random_device rand{};
+        assert(rand.max() - rand.min() >= 256);
+        for(int i=0; i < 8; i++) {
+            for (int j=0; j < 12; j++) {
+                aes_key_share.bytes[i*16 + j] = rand();
+            }
+            aes_key_share.words[i*4 + 3] = tmp_key.words[i]
+                                        ^ aes_key_share.words[i*4 + 2]
+                                        ^ aes_key_share.words[i*4 + 1]
+                                        ^ aes_key_share.words[i*4];
```

Comment:
It probably doesn't matter, but this appears to be doing `D[i] = X[i] ^ C[i] ^ B[i] ^ A[i]` which is different to the `D[i] = X[i] ^ A[i] ^ B[i] ^ C[i]` described in the README?

Reply:
I've fixed the order in the c++ anyway